### PR TITLE
Lga 2237 new backend instance for out of hours

### PR DESCRIPTION
**Pull request by TawneeOwl**
## What does this pull request do?

Adds in a new OOH setting for the end to end tests to test completing a case both in and out of rota hours

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- https://dsdmoj.atlassian.net/jira/software/c/projects/LGA/boards/226?modal=detail&selectedIssue=LGA-2237
- Added a new user called CHS_GENERAL_USER_OOH
- Created two new envs for frontend and backend OOH
- Wrote new OOH tests for complete case
- Set up env variables for in and out of hours for the cla_backend

### CODE REVIEW

File: behave/docker-compose.yml [hunk 1]
```diff
@@ -148,6 +162,29 @@ services:
...
+    <<: *clabackend
+    ports:
+      - "8014:8000"
+    depends_on:
+      - clabackend
+    container_name:
+      clabackendooh
+    environment:
+      <<: *clabackendenvironment
+      NON_ROTA_START_TIME_HR: 22
```
Comment (jogilder) [inline behave/docker-compose.yml:182]: what happens if these tests run at 23:30?

File: behave/docker-compose.yml [hunk 2]
```diff
@@ -148,6 +162,29 @@ services:
       STATIC_FILES_BACKEND: local
       ADMIN_PASSWORD: cla_admin
       ALLOWED_HOSTS: "*"
+      NON_ROTA_START_TIME_HR: 1
```
Comment (jogilder) [inline behave/docker-compose.yml:165]: what happens if these tests run after 22:00? If it is 23:00, is this then out of hours?

File: behave/docker-compose.yml (hunk 1 above)
Comment (jogilder) [inline behave/docker-compose.yml:182]: Is there an option for this to be in hours when there is an hour between start and end? Can we make it so they are the same?

File: behave/features/steps/frontend_assignment.py [hunk 3]
```diff
@@ -494,6 +513,24 @@ def wait_until_dashboard_page_is_loaded(*args):
     wait.until(wait_until_dashboard_page_is_loaded)
 
 
+@step("the case does not show up on the call centre dashboard ooh")
+def step_impl_case_removed_from_list_ooh(context):
+    dashboard_url = "http://clafrontendooh:8000/call_centre/?ordering=-modified&page=1"
```
Comment (jogilder) [inline behave/features/steps/frontend_assignment.py:518]: maybe declare CLA_FRONTEND_OOH_URL in constants or wherever this variable is declared?

File: behave/helper/constants.py [hunk 4]
```diff
@@ -29,6 +29,13 @@
         "user_type": "OPERATOR",
         "application": "FRONTEND",
     },
+    "CHS_GENERAL_USER_OOH": {
+        "username": "test_operator",
+        "password": "test_operator",
+        "login_url": "http://clafrontendooh:8000/auth/login/",
```
Comment (jogilder) [inline behave/helper/constants.py:35]: declare url in constants

File: behave/docker-compose.yml (hunk 2 above)
Comment (TawneeOwl) [reply behave/docker-compose.yml:165]: Yeah it'll classify it out of hours, there is no way around this without re-writing CLA_Backend as the datetime is preset to accept 0 minutes so we cannot make the end time 23:58 to 23:59, 23 is the latest as datetime won't accept 24

File: behave/docker-compose.yml (hunk 1 above)
Comment (TawneeOwl) [reply behave/docker-compose.yml:182]: There is no way around this without re-writing CLA_Backend as the datetime is preset to accept 0 minutes so we cannot make the end time 23:58 to 23:59, 23 is the latest as datetime won't accept 24

File: behave/features/steps/frontend_assignment.py (hunk 3 above)
Comment (TawneeOwl) [reply behave/features/steps/frontend_assignment.py:518]: Amended

File: behave/helper/constants.py (hunk 4 above)
Comment (TawneeOwl) [reply behave/helper/constants.py:35]: Amended

File: behave/docker-compose.yml (hunk 1 above)
Comment (jogilder) [reply behave/docker-compose.yml:182]: Make this set be an hour in the middle of the night (0 to 1 or perhaps 1 till 2) - the least likely time for the test to be run.

File: behave/docker-compose.yml (hunk 2 above)
Comment (jogilder) [reply behave/docker-compose.yml:165]: can we at least make it 0 to 23?

**Review by jogilder — APPROVED** [review]
LGTM

**Review by Pheonnexx — APPROVED** [review]
LGTM 👍

**Pull request merged**